app/models/ext_management_system.rb

```diff
@@ -260,6 +264,78 @@ def default_endpoint
     default || endpoints.build(:role => "default")
   end
 
+  # TODO: takes multiple connection data
+  # endpoints, authentications, and works out what's what.
+  def connections=(options)
+    options.each do |option|
+      endpoint = endpoint_builder(option[:endpoint])
+      authentication_builder(option[:authentication])
```

Comment:
It looks like endpoint and authentication always come in pairs...I see a similar pattern of `endpoint_builder` followed by `authentication_builder` down below.  Should this pair be extracted into a method like `build_connection` or `ensure_connection`?  This might also help with my naming concerns below.